Modules/_functoolsmodule.c

```diff
@@ -1203,6 +1205,8 @@ lru_cache_new(PyTypeObject *type, PyObject *args, PyObject *kw)
     obj->func = func;
     obj->misses = obj->hits = 0;
     obj->maxsize = maxsize;
+    obj->kwd_mark = state->kwd_mark;                        // Borrowed
+    obj->lru_list_elem_type = state->lru_list_elem_type;    // Borrowed
```

Comment:
Would it possible to use a strong reference instead? Maybe the module instance can be deleted but the object survives longer. You need to update lru_cache_tp_traverse() and lru_cache_tp_clear() in this case.

Reply:
Done.